base/compiler/typeinfer.jl

```diff
@@ -389,7 +389,7 @@ function cache_result!(interp::AbstractInterpreter, result::InferenceResult)
     end
 
     if isdefined(result, :escapes)
-        EscapeAnalysis.GLOBAL_ESCAPE_CACHE[linfo]= result.escapes
+        code_cache(interp)[linfo].arg_escapes = result.escapes
```

Comment:
Where is this read? I only see `arg_escapes` on the LHS

Reply:
Yeah it seems to be you need to change `EscapeAnalysis.jl:1222` to actually read this and then delete line 592

Reply:
yes.

And it would be clearer if we just pass `escapes` to the `CodeInstance` constructor.